### ISSUE TITLE
NetInfo not works after wifi become offline -> online

### ISSUE THREAD

**Issue by raminious** [facebook/react-native#10819]
### Description
NetInfo not works after I turn off wifi and then turn in on
### Reproduction

```js
componentDidMount() {
    NetInfo.isConnected.addEventListener('change', this.onConnectionChange)
}

componentWillUnmount() {
    NetInfo.removeEventListener('change', this.onConnectionChange)
}

onConnectionChange = connected =>  {
    console.log('Is connected? ', connected)
}

```

![wifi](https://cloud.githubusercontent.com/assets/334716/20115139/62e76a9c-a60c-11e6-8463-e5f390e8b121.gif)

### Additional Information

* React Native version: 0.36.0
* Platform: iOS 10.1 (Simulator)
* Operating System: MacOS

**Comment by vonovak** [comment]
have you test this on device?

**Comment by raminious** [comment]
No @vonovak. I don't have iPhone

**Comment by okadan** [comment]
same error.

it worked on real device but not on simulator. (iPhone5 / iOS10)

this is my code snippet.

```
constructor(props) {
  super(props);
  this.setState({isConnected: true});
}

componentDidMount = () => {
  NetInfo.isConnected.addEventListener('change', this._handleConnectivity);
}

componentWillUnmount = () => {
  NetInfo.isConnected.removeEventListener('change', this._handleConnectivity);
}

_handleConnectivity = (isConnected) => {
  this.setState({isConnected,})
}

render() {
  return (
    <View>
      <Text>{this.state.isConnected ? 'online' : 'offline'}</Text>
    </View>
  );
}
```

**Comment by vonovak** [comment]
yes, it does work on device too. So my assumption is that RN is not at fault here.

**Comment by raminious** [comment]
Thank you @okadan 
Okay @vonovak  
So I close this issue.

**Comment by maslianok** [comment]
So NetInfo doesn't work on simulator, right?

**Comment by RobertSheaO** [comment]
I think I've determined that the issue is when you enable debugging in the browser with the simulator. I've encountered other issues because of debugging in the browser. The app runs the JS through the browser's implementation of JS, rather than the JS implementation running on the device/simulator. This is probably also the root of this bug. I confirm the issue by putting `alert(isConnected)` with debugging disabled, rather than `console.log(isConnected)` with debugging.
